test-support/run-picolibc-test.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023, Arm Limited and affiliates.
+
+# This is a wrapper script to run picolibc tests with qemu
+
+import sys
+import argparse
+import subprocess
+
+disabled_tests = [
+    "picolibc_aarch64-build/test/math_errhandling",
+    "picolibc_armv7em_hard_fpv4_sp_d16-build/test/math_errhandling",
+    "picolibc_armv7em_hard_fpv5_d16-build/test/math_errhandling",
+    "picolibc_armv8m.main_hard_fp-build/test/printf_scanf",
+    "picolibc_armv8m.main_hard_fp-build/test/printff_scanff",
+    "picolibc_armv8m.main_hard_fp-build/test/printff-tests",
+    "picolibc_armv8m.main_hard_fp-build/test/math_errhandling",
+    "picolibc_armv8m.main_hard_fp-build/test/rounding-mode",
+    "picolibc_armv8m.main_hard_fp-build/test/long_double",
+    "picolibc_armv8m.main_hard_fp-build/test/rand",
+    "picolibc_armv8m.main_hard_fp-build/test/fenv",
+    "picolibc_armv8m.main_hard_fp-build/test/math-funcs",
+    "picolibc_armv8m.main_hard_fp-build/test/test-strtod",
+    "picolibc_armv8m.main_hard_fp-build/test/test-efcvt",
+    "picolibc_armv8m.main_hard_fp-build/test/complex-funcs",
+    "picolibc_armv8m.main_hard_fp-build/test/semihost/semihost-times",
+    "picolibc_armv8m.main_hard_fp-build/newlib/libm/test/math_test",
+    "picolibc_armv8m.main_hard_fp-build/test/libc-testsuite/sscanf",
+    "picolibc_armv8m.main_hard_fp-build/test/libc-testsuite/strtod",
+    "picolibc_armv8.1m.main_soft_nofp_nomve-build/newlib/libm/test/math_test",
+    "picolibc_armv8.1m.main_hard_fp-build/test/math_errhandling",
+    "picolibc_armv8.1m.main_hard_fp-build/newlib/libm/test/math_test",
+    "picolibc_armv8.1m.main_hard_nofp_mve-build/test/fenv",
+    "picolibc_armv8.1m.main_hard_nofp_mve-build/newlib/libm/test/math_test",
+    "picolibc_armv8.1m.main_hard_nofp_mve-build/test/math_errhandling"
+]
+
+def is_disabled(image):
+    return any([image.endswith(t) for t in disabled_tests])
+
+def run(args):
+    """Execute the program using QEMU and return the subprocess return code."""
+    command = args.command
+    assert command
+    image = command[0]
+    image_args = command[1:]
+    if image_args[0:1] == ['--']:
+        image_args = image_args[1:]
+    if is_disabled(image):
+        return 77
+    semihosting =  ','.join(
+        ["enable=on", "chardev=stdio0"]
+        + ['arg=' + arg.replace(',', ',,') for arg in image_args])
+    qemu_params = args.qemu_params.split(':') + [
+            "-chardev", "stdio,mux=on,id=stdio0",
+            "-semihosting-config", semihosting,
+            "-monitor", "none",
+            "-serial", "none",
+            "-nographic"]
+    if qemu_params.count("virt"):
```

Comment:
Rather than trying to parse the qemu parameters, pull the machine type out into a separate argument that is parsed with argparse.

Reply:
Done in https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/pull/256